### PR TITLE
feat: Collection Performance Testing and Optimization

### LINKED ISSUE #8: Performance: Add benchmarks for Persistent Collections to ensure that all the methods work as expected with 10k+ records
Here is the test suite for the reference:

https://github.com/near/near-sdk-rs/blob/master/near-sdk/tests/store_performance_tests.rs

Consider making the suite SDK-agnostic.

Since we currently don't have near-workspaces-py, I would suggest to either take the existing near-workspaces-rs implementation from near-sdk-rs, or port it to near-workspaces-js.

### PR DESCRIPTION
**Pull request by r-near**
## Overview

This PR adds comprehensive performance testing for all collection types in the NEAR SDK Python and implements key optimizations for `UnorderedMap` and `UnorderedSet` to improve gas efficiency.

## Key Changes

1. **Performance test framework**:
   - Added integration tests for measuring gas consumption of common operations
   - Created testing infrastructure to benchmark collections with 10,000 elements
   - Generated Markdown reports with detailed performance metrics

2. **UnorderedMap/UnorderedSet optimizations**:
   - Implemented O(1) removal operations using index tracking
   - Added efficient swap-remove implementation for better gas usage
   - Reduced gas costs for high-frequency operations

3. **Documentation**:
   - Added performance benchmark results for each collection type
   - Documented relative efficiency of operations compared to baseline
   - Included testing methodology details

## Performance Results

All collections were tested with 10,000 elements:

- **LookupMap/LookupSet**: Extremely efficient for lookups (1.00x baseline)
- **Vector**: Consistent performance for indexed access (1.07x baseline)
- **UnorderedMap/UnorderedSet**: Efficient for iterable collections (most operations under 1.3x baseline)

The most expensive operations are removes in UnorderedMap (~1.62x baseline) due to additional index management, but still significantly better than before the optimizations.

## Implementation Details

The core optimization involves tracking indices for each key/value in UnorderedMap/UnorderedSet, allowing O(1) removal operations instead of O(n) linear searches. Key features:

- Index mapping stored in contract storage with a new prefix
- Efficient swap-remove operations that maintain index integrity
- Graceful handling of edge cases (e.g., removing the last element)

## Testing

The PR includes:
- Unit tests for the optimized implementations
- Integration tests measuring gas consumption
- Comparative benchmarks against baseline operations

## Next Steps

Future work could include:
- TreeMap implementation and performance optimization (to be addressed in a separate PR)
- Additional optimizations for batch operations
- Exploring ways to reduce storage costs
- Implementing lazy deletion strategies for further gas savings

Closes #8 

@race-of-sloths

### CODE REVIEW

**Comment by race-of-sloths** [comment]
🥁  __Score it!__

@r-near, please ask your peers to score the PR using `@race-of-sloths score [1/2/3/5/8/13]`.
If no scoring is provided within 24 hours, this PR will be scored as 1 🦥

File: src/near_sdk_py/decorators.py [hunk 1]
```diff
@@ -21,18 +21,24 @@ def contract_method(func):
...
+            )
             # If no kwargs were provided and this appears to be a blockchain call
             if len(kwargs) == 0 and len(args) <= 1:
                 try:
                     data = Input.string()
+                    Log.debug(f"Input data: {data}")
                     if len(data) > 0:
                         # Try parsing it as JSON
                         kwargs = json.loads(data)
+                        Log.debug(f"Parsed kwargs: {kwargs}")
```
Comment (frol) [inline src/near_sdk_py/decorators.py:35]: These are verbose and make unnecessary host-function calls (gas usage + storage), so I'd not enable these by default for all SDK users.

Comment (r-near) [reply src/near_sdk_py/decorators.py:35]: Resolved with #26
